app/code/Magento/ConfigurableProduct/Test/Unit/Ui/DataProvider/Product/ConfigurableDataProviderTest.php

```diff
@@ -0,0 +1,122 @@
+ */
+class ConfigurableDataProviderTest extends TestCase
+{
+    /**
+     * @var ObjectManager
+     */
+    protected $objectManager;
+
+    /**
+     * @var CollectionFactory|MockObject
+     */
+    protected $collectionFactoryMock;
+
+    /**
+     * @var Collection|MockObject
+     */
+    protected $collectionMock;
+
+    /**
+     * @var PoolInterface|MockObject
+     */
+    private $modifiersPool;
+
+    /**
+     * @var ModifierInterface|MockObject
+     */
+    private $modifierMockOne;
+
+    /**
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        $this->objectManager = new ObjectManager($this);
+
+        $this->collectionMock = $this->getMockBuilder(Collection::class)
+            ->disableOriginalConstructor()
+            ->setMethods(
+                [
+                    'toArray',
+                    'isLoaded',
+                    'addAttributeToFilter',
+                    'load',
+                    'getSize',
+                    'addFilterByRequiredOptions',
+                    'addStoreFilter'
+                ]
+            )->getMock();
+        $this->collectionFactoryMock = $this->getMockBuilder(CollectionFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+        $this->collectionFactoryMock->expects($this->any())
+            ->method('create')
+            ->willReturn($this->collectionMock);
+
+        $this->modifiersPool = $this->getMockBuilder(PoolInterface::class)
+            ->getMockForAbstractClass();
+        $this->modifierMockOne = $this->getMockBuilder(ModifierInterface::class)
+            ->setMethods(['modifyData'])
+            ->getMockForAbstractClass();
+        $this->modifierMockOne->expects($this->any())
+            ->method('modifyData')
+            ->willReturn($this->returnArgument(0));
+        $this->modifiersPool->expects($this->any())
+            ->method('getModifiersInstances')
+            ->willReturn([$this->modifierMockOne]);
+    }
+
+    /**
+     * @return object
+     */
+    protected function getModel()
+    {
+        return $this->objectManager->getObject(ConfigurableDataProvider::class, [
+            'name' => 'testName',
+            'primaryFieldName' => 'testPrimaryFieldName',
+            'requestFieldName' => 'testRequestFieldName',
+            'collectionFactory' => $this->collectionFactoryMock,
+            'meta' => [],
+            'data' => [],
+            'addFieldStrategies' => [],
+            'addFilterStrategies' => [],
+            'modifiersPool' => $this->modifiersPool
+        ]);
+    }
+
+    /**
+     * Test checks ConfigurableDataProvider type
+     */
+    public function testCheckType()
+    {
+        $this->assertInstanceOf(ConfigurableDataProvider::class, $this->getModel());
+    }
+
+    /**
+     * Test checks collection type
+     */
+    public function testGetCollection()
```

Comment:
missing return type

Reply:
Usually it's good practice to put all the public methods first  and then the private ones

Reply:
Fixed by moving public functions first. Then protected then private ones at the end ....